File: 0455-assign-cookies/0455-assign-cookies.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) 
    {
        int c = 0; // Counter for content children
        sort(g.begin(), g.end()); // Sort greed factors
        sort(s.begin(), s.end()); // Sort cookie sizes
        int i = 0, j = 0; // Pointers for greed factors and cookie sizes

        // Traverse both arrays
        while (i < g.size() && j < s.size()) {
            if (s[j] >= g[i]) { // If the cookie can satisfy the child
                c++; // Increment content children count
                i++; // Move to the next child
            }
            j++; // Move to the next cookie
        }

        return c;
    }
};

File: 0455-assign-cookies/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        count = 0 
        g.sort()
        s.sort()
        i = 0
        j = 0
        while(i < len(g) and j <len(s)):
            if(s[j] >= g[i]):
                count = count+1
                i = i+1
            j = j+1
        return count
      